core/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -288,7 +282,7 @@ public final Searcher acquireSearcher(String source) throws EngineException {
             try {
                 final Searcher retVal = newSearcher(source, searcher, manager);
                 success = true;
-                return retVal;
+                return config().getWrappingService().wrap(retVal);
```

Comment:
unfortunately EngineSearcherFactory is only used once per refresh and not invoked each time a search is happening. The IndexSearcherWrapperService does delegate to the close method of the original Engine.Searcher do the ref counting should add up.